cli/internal/packagemanager/packagemanager.go

```diff
@@ -36,6 +36,10 @@ type PackageManager struct {
 	// The directory in which package assets are stored by the Package Manager.
 	PackageDir string
 
+	// The separator that the Package Manger uses to identify arguments that
+	// should be passed through to the underlying script.
+	ArgSeparator []string
```

Comment:
I know this maps to the previous, but can this be changed to not be an array of strings?

Reply:
What's the most Go-like way to represent this? The comment on the pnpm 6/7 split:
> // We are allowed to use nil here because argSeparator already has a type, so it's a typed nil,
> // vs if we tried to call "args = append(args, nil...)", which would not work. This could
> // just as easily be []string{}, but the style guide says to prefer nil for empty slices.

Would it be best practice/common to use the empty string as the nil value and check it before appending to the arg slice?

Reply:
The existing solution is too clever by half, but I understand now why it exists.

- String types can't be `nil` (only empty string)
- `[]string{}`'s "zero value" is `nil`.

I'm ... comfortable leaving it the way it is. 🤷‍♂️ The only other reasonable options are:
- check for "" (not ideal since it's an in-band sentinel)
- far more-invasive (a whole new struct just for arg separator, and a method to get it, guards at the `append`, and then it's all just a mess)

So, yeah, let's add more to the comment and call it done.